zetaclient/bitcoin_client.go

```diff
@@ -586,20 +605,33 @@ func (ob *BitcoinChainClient) fetchUTXOS() error {
 	return nil
 }
 
-func (ob *BitcoinChainClient) findNonceMarkUTXO(nonce uint64, tssAddress string) (int, error) {
-	outTxID := ob.GetTxID(nonce)
-	res, mined := ob.includedTx[outTxID]
-	if !mined {
-		return -1, fmt.Errorf("findNonceMarkUTXO: outTx %s not included yet", outTxID)
+func (ob *BitcoinChainClient) findTxIDByNonce(nonce uint64, test bool) (string, error) {
+	ob.mu.Lock()
+	res, included := ob.includedTxResults[ob.GetTxID(nonce)]
+	ob.mu.Unlock()
+
+	if included {
+		return res.TxID, nil
+	} else if test {
+		return "", fmt.Errorf("findTxIDByNonce: error getting cctx for nonce %d", nonce)
+	} else {
+		send, err := ob.zetaClient.GetCctxByNonce(ob.chain.ChainId, nonce)
+		if err != nil {
+			return "", errors.Wrapf(err, "findTxIDByNonce: error getting cctx for nonce %d", nonce)
+		}
+		return send.GetCurrentOutTxParam().OutboundTxHash, nil
 	}
```

Comment:
imo this is not very readable , would you conside using a switch case on `test` , for `!included`